base_ebill_payment_contract/models/ebill_payment_contract.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class EbillPaymentContract(models.Model):
+
+    _name = "ebill.payment.contract"
+    _description = "Ebill Payment Contract"
+    _rec_name = "name"
+
+    transmit_method_id = fields.Many2one(
+        comodel_name="transmit.method",
+        string="Service Name",
+        ondelete="restrict",
+    )
+    partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        required=True,
+        string="Customer"
+    )
+    name = fields.Char(related="partner_id.name")
+    date_start = fields.Date(
+        string="Start date",
+        required=True,
+        default=fields.Date.today,
+    )
+    date_end = fields.Date(string="End date")
+    state = fields.Selection(
+        selection=[("draft", "Draft"), ("open", "Open"), ("cancel", "Cancel")],
+        required=True,
+        default='draft',
+    )
+    is_valid = fields.Boolean(compute="_compute_is_valid", store=True)
+
+    @api.onchange('state')
+    def _compute_state_changed(self):
+        """Change the end date if contract is canceled."""
+        if self.state == 'cancel' and self.date_end > fields.Date.today():
+            self.date_end = fields.Date.today()
+
+    @api.multi
+    @api.depends("state", "date_start", "date_end")
```

Comment:
I'm just wondering here, if `is_valid` was computed at record creation, and after a while `date_end` is yesterday, the field won't be recomputed since there is no trigger, is it right ?

Reply:
This is a good point. Maybe `is_valid` should not be a field then but simply a function ?

Reply:
If you remove `store=True` it will be recomputed anyway so there wouldn't be any issue. But if it must be stored, I don't know what's the best practice to recompute the field everyday.